### ISSUE TITLE
Each transaction is market as an error on New Relic

### ISSUE THREAD

**Issue by Kolesar** [Seldaek/monolog#1218]
I do not know what is the initial idea when this code is written, but with this `else` condition here each transaction will be marked as an error.

https://github.com/Seldaek/monolog/blob/6bb347c040bb29f42c2c249714aac8c0e36c2454/src/Monolog/Handler/NewRelicHandler.php#L94

**Comment by Kolesar** [comment]
To be honest I do not have experience with New Relic too. My suggestion is only what I found during implementation it in my project, but probably it is the absolutely wrong way. As you say, maybe a different way of thinking and implementation is the solution.

**Comment by jderusse** [comment]
Looking at the comment https://github.com/Seldaek/monolog/pull/1217#issuecomment-436212898 I think there is a misunderstood of the usage of new `NewRelicHandler` (or more generally a misunderstood of NewRelic)

NewRelic is not made for tracking logs (like logmatic or Sentry, ...), it only contains API to trac errors and response time.

The purpose of this handler is to send application's dis-function logs to NewRelic. In order to NOT send debug, notice or informational logs, you shouldn't configure a log level too low (IMHO `error` is enougth)

```
        newrelic:
            type: newrelic
            level: error
```

**Comment by Seldaek** [comment]
@jderusse I'd tend to agree there. Do you have any opinion on https://github.com/Seldaek/monolog/pull/1217#issuecomment-436189051 ? Is that the correct interpretation? Relating to https://github.com/Seldaek/monolog/blob/master/src/Monolog/ResettableInterface.php#L15

**Comment by jderusse** [comment]
@Seldaek I answered in the PR (see https://github.com/Seldaek/monolog/pull/1217#issuecomment-446104846).

TL;DR: this is not the responsibility of Monolog to start or end a NewRelic transaction

**Comment by jamesmehorter** [comment]
> NewRelic is not made for tracking logs

This is no longer true, NewRelic does now track logs. The NewRelic PHP Agent captures application errors—I would expect this logging package's "NewRelicHandler" to solely ship logs to NR.

**Comment by Seldaek** [comment]
Ok so what I'm hearing is we maybe should just deprecate this handler? Or is there still value in the error collection?

**Comment by jamesmehorter** [comment]
It seems that New Relic's PHP Agent presently does not log meta data which has been appended to log entries—only the log message is sent (whereas the Java Agent does do this). Perhaps once that feature is available deprecation of this handler would be logical.
